### DISCUSSION TITLE
auth.getUser not working as expected with or without jwt token

### DISCUSSION THREAD

**Discussion by sarinthecoder** [supabase/supabase#19621] [Questions] [answered]
![image](https://github.com/supabase/supabase/assets/96570513/cc4e2325-0e10-4062-8933-72b72e01a809)
In the above code , I am trying to get the user who is logged in or get the user by passing jwt.  I am getting undefined .. But the same JWT token works while I call using POSTMAN "xxxxxx.supabase.co/auth/v1/user" call. 
In the edge function I get undefined.
![image](https://github.com/supabase/supabase/assets/96570513/385340e5-5345-47ab-a641-80fa17d889fc)

Not able to understand what is wrong here ; Why does it work using curl call but not using the auth.getUser() call . It seemed to be working as expected yesterday.

**Comment by GaryAustin1** [comment, marked as answer]
You need to use user not user1 in your return.  You have to use the name of the real name of the return object which is user and session in data. 
You can alias the return name like `user:user1`

**Reply by sarinthecoder** [reply]
Thanks

**Reply by CamelliaGh** [reply]
Thank you! You really saved me!

**Reply by aelouadi** [reply]
Thanks a lot
